tests/expected_tests.cc

```diff
@@ -0,0 +1,37 @@
+#include "gtest/gtest.h"
+#include "stout/expected.h"
+
+stout::expected<int> divide(const int dividend, const int divisor) {
+  if (divisor == 0) {
+    return stout::make_unexpected("divide by zero");
+  }
+
+  return dividend / divisor;
+}
+
+TEST(Expected, SucceedResult) {
+  const stout::expected<int> result = divide(6, 3);
+
+  EXPECT_EQ(result, stout::expected<int>(2));
+}
+
+TEST(Expected, DivideByZero) {
+  const stout::expected<int> result = divide(6, 0);
+
+  EXPECT_EQ(result, stout::make_unexpected("divide by zero"));
+}
+
+TEST(Expected, MakeUnexpected) {
+  auto error = []() {
```

Comment:
Do you even need the lambda in this test case, since the new definition of `make_unexpected` is no longer doing return type inference?

`auto unexpected = stout::make_unexpected("error");`

Reply:
Yep, I think we do not need this lambda!